src/day8/re_frame/trace.cljs

```diff
@@ -144,20 +144,27 @@
     (fn [trace]
       (< (:query query) (:duration trace)))))
 
-(defn render-traces [showing-traces filter-items trace-detail-expansions]
+(defn add-filter [filter-items filter-input filter-type from-mouse?]
+  (swap! filter-items conj {:id (random-uuid)
+                            :query (if from-mouse?
+                                     (str/lower-case (name filter-input))
+                                     (if (= @filter-type :contains)
```

Comment:
Why is `filter-type` derefed here? Isn't it just a value? If it can sometimes be an atom I think it's simpler to deref it in the call to this function rather than inside it.

Reply:
yes, that makes sense, thank you!